### PR TITLE
Support MongoDB connection type for AWS Glue

### LINKED ISSUE #13007: Support for Glue Connection Type MongoDB
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Glue has recently added MongoDB support for connections.

### Affected Resource

* aws_glue_connection

### Potential Terraform Configuration

```hcl
resource "aws_glue_connection" "example" {
  connection_type = "MONGODB"
}
```

### References

- [Announcement](https://aws.amazon.com/about-aws/whats-new/2020/04/aws-glue-now-supports-reading-from-amazon-documentdb-and-mongodb-tables/)
- [Glue Developer Guide](https://docs.aws.amazon.com/glue/latest/dg/connection-using.html)
- [Glue API Reference](https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-catalog-connections.html)
- [Terraform AWS Provider Contributing Guide](https://github.com/terraform-providers/terraform-provider-aws/blob/master/.github/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by drexler**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #13007

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
support MongoDB connection type for AWS Glue
```

Output from acceptance testing:

```
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSGlueConnection_MongoDB'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSGlueConnection_MongoDB -timeout 120m
=== RUN   TestAccAWSGlueConnection_MongoDB
=== PAUSE TestAccAWSGlueConnection_MongoDB
=== CONT  TestAccAWSGlueConnection_MongoDB
--- PASS: TestAccAWSGlueConnection_MongoDB (15.14s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	15.208s


...
```

### CODE REVIEW

File: aws/resource_aws_glue_connection_test.go [hunk 1]
```diff
@@ -439,3 +472,19 @@ resource "aws_glue_connection" "test" {
 }
 `, rName)
 }
+
+func testAccAWSGlueConnectionConfig_MongoDB(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_glue_connection" "test" {
```
Comment (anGie44) [inline aws/resource_aws_glue_connection_test.go:478]: small nit-pick here to align this config with `terrafmt fmt`

**Comment by drexler** [comment]
Thanks for the review @anGie44. I've updated the PR and added the documentation snippet.

>I think it'd be helpful to document the newly supported connection_type in glue_connection.html.markdown (e.g. "Supported: JDBC, SFTP, MONGODB")

Interestingly enough SFTP is [**not supported with the Go SDK**](https://docs.aws.amazon.com/sdk-for-go/api/service/glue/#ConnectionInput) so omitted that from the documentation. Should i file a follow-up issue to remove it from the resource validation check? Alternatively, i could do it as part of this PR - it doesn't seem to be a breaking change to do.

**Comment by anGie44** [comment]
@drexler interesting! hmm seems like the online API docs haven't been updated.. yes, please file a follow-up issue so we have it documented. thanks for investigating :)

**Comment by drexler** [comment]
@anGie44 issue for SFTP connection support created.

**Pull request merged**